static/src/javascripts/projects/commercial/modules/messenger/click.ts

```diff
@@ -0,0 +1,22 @@
+import { isString } from '@guardian/libs';
+import { trackNativeAdLinkClick } from '../../../common/modules/analytics/google';
+import type { RegisterListener } from '../messenger';
+
+const sendClick = (adSlot: { id: string }, linkName: string): void => {
+	trackNativeAdLinkClick(adSlot.id, linkName);
+};
+
+const init = (register: RegisterListener): void => {
+	register('click', (linkName, ret, iframe) =>
+		sendClick(
+			iframe?.closest('.js-ad-slot') ?? {
+				id: 'unknown',
+			},
+			isString(linkName) ? linkName : '',
+		),
+	);
+};
+
+export { init };
+
+export const _ = { sendClick };
```

Comment:
Another pattern I think is a bit iffy, exporting what-should-be private methods to test them?

This is a TS conversion so totally minor and optional... I'm just pointing at things :)

Reply:
Agreed about iffiness - for you is it because we shouldn't be testing un-exported functions at all, since they are necessarily don't have an external interface? Or do you just not like the `export const _ =` pattern and think we should just `export` as normal?

Reply:
Exporting functions to facilitate testing - Ideally I think we should be treating the subject-under-test as a black box. This also makes the test less brittle to implementation changes.

On exporting as `_` . I am not against it if there's a valid reason to distinguish some types of exports from other - e.g. for `init` type functions, although even that is debateable. However to just use it to denote internal methods for testing, imo we shouldn't be exporting those in the first place.

Reply:
Well there's no reason we can't just test `init`, I've updated the tests so we don't need to export `_` then.